From 10773687163a84ba6be4440039a237c2b901e93c Mon Sep 17 00:00:00 2001
From: Tamas Nepusz <ntamas@gmail.com>
Date: Tue, 23 Feb 2021 16:16:08 +0100
Subject: [PATCH] feat: allow external f2c arith.h headers when cross-compiling


diff --git a/vendor/f2c/CMakeLists.txt b/vendor/f2c/CMakeLists.txt
index 52c573ffa..f7b5107dd 100644
--- a/vendor/f2c/CMakeLists.txt
+++ b/vendor/f2c/CMakeLists.txt
@@ -5,14 +5,25 @@ if (NOT MSVC)
   target_link_libraries(arithchk PRIVATE m)
 endif()
 
-add_custom_command(
-  OUTPUT arith.h
-  COMMENT "Generating arith.h for f2c..."
-  COMMAND arithchk > ${CMAKE_CURRENT_BINARY_DIR}/arith.h
-  DEPENDS arithchk
-  VERBATIM
+# Provide an option for the user to provide an external arith.h for
+# cross-compilation
+set(
+  F2C_EXTERNAL_ARITH_HEADER "" CACHE FILEPATH
+  "Path to an external arith.h to use for compiling f2c, typically for cross-compilation"
 )
 
+if(F2C_EXTERNAL_ARITH_HEADER)
+  configure_file(${F2C_EXTERNAL_ARITH_HEADER} arith.h COPYONLY)
+else()
+  add_custom_command(
+    OUTPUT arith.h
+    COMMENT "Generating arith.h for f2c..."
+    COMMAND arithchk > ${CMAKE_CURRENT_BINARY_DIR}/arith.h
+    DEPENDS arithchk
+    VERBATIM
+  )
+endif()
+
 # Declare the files needed to compile our vendored f2c copy
 add_library(
   f2c_vendored
