diff --git a/etc/cmake/test_helpers.cmake b/etc/cmake/test_helpers.cmake
index 693ba3c28..9602834cc 100644
--- a/etc/cmake/test_helpers.cmake
+++ b/etc/cmake/test_helpers.cmake
@@ -70,7 +70,7 @@ function(add_legacy_test FOLDER NAME NAMESPACE)
 
     # Semicolons are used as list separators in CMake so we need to escape them in the PATH,
     # otherwise the PATH envvar gets split by CMake before it passes the PATH on to CTest.
-    string(JOIN "\;" CORRECT_PATH $ENV{PATH})
+    string(JOIN "\\\;" CORRECT_PATH $ENV{PATH})
     SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES ENVIRONMENT "PATH=${IGRAPH_LIBDIR}\;${CORRECT_PATH}" )
   endif()
 endfunction()
