diff --git a/etc/cmake/test_helpers.cmake b/etc/cmake/test_helpers.cmake
index 693ba3c28..e231bd860 100644
--- a/etc/cmake/test_helpers.cmake
+++ b/etc/cmake/test_helpers.cmake
@@ -62,16 +62,24 @@ function(add_legacy_test FOLDER NAME NAMESPACE)
     # so that igraph.dll is found when running the tests.
     SET(IGRAPH_LIBDIR $<TARGET_FILE_DIR:igraph>)
 
-    # The next line is necessitated by MinGW on Windows. MinGW uses forward slashes in
-    # IGRAPH_LIBDIR, but we need to supply CTest with backslashes because CTest is executed
-    # in a cmd.exe shell. So we simply replace forward slashes with backslases in
-    # IGRAPH_LIBDIR.
-    string(REPLACE "/" "\\" IGRAPH_LIBDIR ${IGRAPH_LIBDIR})
+    SET(TEST_PATHS ${IGRAPH_LIBDIR})
+    list(APPEND TEST_PATHS $ENV{PATH})
+    SET(TEST_CORRECT_PATHS)
+    foreach (PATH ${TEST_PATHS})
+      MESSAGE(STATUS "Original path: ${PATH}")
+      file(TO_NATIVE_PATH "${PATH}" CORRECT_PATH)
+      # Remove trailing backslashes because this may interfere with the interpretation
+      # as an escape character
+      STRING(REGEX REPLACE "\\$" "" CORRECT_PATH ${CORRECT_PATH})
+      list(APPEND TEST_CORRECT_PATHS ${CORRECT_PATH})
+      MESSAGE(STATUS "Added path: ${CORRECT_PATH}")
+    endforeach()
 
     # Semicolons are used as list separators in CMake so we need to escape them in the PATH,
     # otherwise the PATH envvar gets split by CMake before it passes the PATH on to CTest.
-    string(JOIN "\;" CORRECT_PATH $ENV{PATH})
-    SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES ENVIRONMENT "PATH=${IGRAPH_LIBDIR}\;${CORRECT_PATH}" )
+    string(JOIN "\;" CORRECT_PATHS ${TEST_CORRECT_PATHS})
+    MESSAGE(STATUS "CORRECT_PATHS: ${CORRECT_PATHS}")
+    SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES ENVIRONMENT "PATH=${CORRECT_PATHS}" )
   endif()
 endfunction()
 
